include/lsqecc/layout/ascii_layout_spec.hpp

```diff
@@ -130,6 +134,8 @@ class LayoutFromSpec : public Layout{
     std::vector<SurfaceCodeTimestep> cached_distillation_times_;
     std::vector<Cell> cached_ancilla_locations_;
     std::vector<std::vector<Cell>> cached_distilled_state_locations_;
+    std::vector<Cell> cached_dead_cells_;
+    bool magic_states_reserved_; 
```

Comment:
No need to cache this here since you can readily compute it as ``bool magic_states_reserved() const override {return cached_distilled_state_locations_.size() > 0;}

Reply:
cached_distilled_state_locations will be populated regardless of whether 'M' cells were specified in the layout spec.